src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -413,43 +429,78 @@ def make_dependency_files(parsed_config, config_file_path, to_stdout):
                             else:
                                 raise ValueError(
                                     f"No matching matrix found in '{include}' for: {matrix_combo}"
                                 )
 
                 # Dedupe deps and print / write to filesystem
                 full_file_name = get_filename(file_type, file_key, matrix_combo)
-                deduped_deps = dedupe(dependencies)
 
-                output_dir = (
-                    "."
-                    if to_stdout
-                    else get_output_dir(file_type, config_file_path, file_config)
-                )
-                contents = make_dependency_file(
+                output_dir = get_output_dir(file_type, config_file_path, file_config)
+
+                output_file_path = os.path.join(output_dir, full_file_name)
+
+                if results.get(output_file_path, None) is None:
+                    if file_type != str(OutputTypes.PYPROJECT):
+                        results[output_file_path] = ""
+                    else:
+                        # pyproject.toml needs to be modified in place instead of built from scratch.
+                        with open(output_file_path) as f:
+                            results[output_file_path] = tomlkit.load(f)
+                            # Append `-cuXX` to `[package.name]`
+                            results[output_file_path]["project"][
+                                "name"
+                            ] = name_with_cuda_suffix(
+                                results[output_file_path]["project"]["name"],
+                                matrix_combo.get("cuda", None),
+                                cuda_suffix,
+                            )
+
+                results[output_file_path] = make_dependency_file(
+                    results[output_file_path],
                     file_type,
                     full_file_name,
-                    config_file_path,
-                    output_dir,
                     channels,
-                    deduped_deps,
+                    dedupe(dependencies),
                     extras,
                 )
 
-                if to_stdout:
-                    print(contents)
-                else:
-                    os.makedirs(output_dir, exist_ok=True)
-                    file_path = os.path.join(output_dir, full_file_name)
-                    with open(file_path, "w") as f:
-                        if file_type == str(OutputTypes.PYPROJECT):
-                            tomlkit.dump(contents, f)
-                        else:
-                            f.write(contents)
+    def write_output(data, output_dir, f):
+
+        relpath_to_config_file = os.path.relpath(config_file_path, output_dir)
+        header = textwrap.dedent(
+            f"""\
+            {HEADER}
+            # To make changes, edit {relpath_to_config_file} and run `{cli_name}`.
+            """
+        )
+
+        if isinstance(data, dict):
```

Comment:
Is this implicitly relying on pyproject outputs generating a dict here while other outputs just contain a long string of text? We should make that condition explicit if so, otherwise it's confusing why tomlkit is getting used in a generic dict path.